compiler/src/test/java/com/squareup/anvil/compiler/codegen/ContributesSubcomponentHandlerGeneratorTest.kt

```diff
@@ -1060,9 +1083,128 @@ class ContributesSubcomponentHandlerGeneratorTest {
+    val firstResult = compile(
+      """
+        package com.squareup.test
+
+        import com.squareup.anvil.annotations.ContributesSubcomponent
+        import com.squareup.anvil.annotations.ContributesTo
+        import com.squareup.anvil.annotations.MergeComponent
+        import dagger.Module
+        import dagger.Provides
+
+        @ContributesSubcomponent(Any::class, parentScope = Unit::class)
+        interface SubcomponentInterface {
+          fun integer(): Int
+        }
+
+        @ContributesTo(Any::class)
+        @Module
+        object DaggerModule1 {
+          @Provides fun provideIntegerFive(): Int = 5
+        }
+
+        @MergeComponent(Unit::class)
+        interface ComponentInterface1
+      """,
+      enableDaggerAnnotationProcessor = true
+    ) {
+      val daggerComponent = componentInterface1.daggerComponent.declaredMethods
+        .single { it.name == "create" }
+        .invoke(null)
+
+      val subcomponent = componentInterface1.methods
+        .single()
+        .invoke(daggerComponent)
+
+      val int = subcomponent::class.java.declaredMethods
+        .single { it.name == "integer" }
+        .use { it.invoke(subcomponent) as Int }
+
+      assertThat(int).isEqualTo(5)
+    }
+
+    compile(
+      """
+        package com.squareup.test
+
+        import com.squareup.anvil.annotations.ContributesSubcomponent
+        import com.squareup.anvil.annotations.ContributesTo
+        import com.squareup.anvil.annotations.MergeComponent
+        import dagger.Module
+        import dagger.Provides
+
+        @ContributesTo(Any::class, replaces = [DaggerModule1::class])
+        @Module
+        object DaggerModule2 {
+          @Provides fun provideIntegerSix(): Int = 6
+        }
+
+        @MergeComponent(Unit::class)
+        interface ComponentInterface2
+      """,
+      enableDaggerAnnotationProcessor = true,
+      previousCompilationResult = firstResult
+    ) {
+      val daggerComponent = componentInterface2.daggerComponent.declaredMethods
+        .single { it.name == "create" }
+        .invoke(null)
+
+      val subcomponent = componentInterface2.methods
+        .last()
+        .invoke(daggerComponent)
+
+      val int = subcomponent::class.java.declaredMethods
+        .single { it.name == "integer" }
+        .use { it.invoke(subcomponent) as Int }
+
+      assertThat(int).isEqualTo(6)
+    }
+  }
+
+  @Test fun `contributed subcomponent parent interfaces are merged with the right component`() {
+    compile(
+      """
+      package com.squareup.test
+
+      import com.squareup.anvil.annotations.ContributesSubcomponent
+      import com.squareup.anvil.annotations.MergeComponent
+
+      @ContributesSubcomponent(Unit::class, parentScope = Any::class)
+      interface SubcomponentInterface
+
+      @MergeComponent(Any::class)
+      interface ComponentInterface1
+
+      @MergeComponent(Any::class)
+      interface ComponentInterface2
```

Comment:
This test highlights the issue as well that had to be solved. Originally, we would have generated only one subcomponent, but have two parent components. That would be fine, Dagger can handle this. But this becomes problematic if there are different bindings in the parent component (that's what the test above is verifying).